hibernate-core/src/main/java/org/hibernate/mapping/Table.java

```diff
@@ -447,8 +447,24 @@ public Iterator sqlAlterStrings(
 		
 		final JdbcEnvironment jdbcEnvironment = metadata.getDatabase().getJdbcEnvironment();
 
+		Identifier quotedCatalog = catalog != null && catalog.isQuoted() ?
+				new Identifier( tableInfo.getName().getCatalogName().getText(), true ) :
+				tableInfo.getName().getCatalogName();
+
+		Identifier quotedSchema = schema != null && schema.isQuoted() ?
+				new Identifier( tableInfo.getName().getSchemaName().getText(), true ) :
+				tableInfo.getName().getSchemaName();
+
+		Identifier quotedTable = name != null &&  name.isQuoted() ?
+				new Identifier( tableInfo.getName().getObjectName().getText(), true ) :
+				tableInfo.getName().getObjectName();
+
 		final String tableName = jdbcEnvironment.getQualifiedObjectNameFormatter().format(
-				tableInfo.getName(),
+				new QualifiedTableName(
+					quotedCatalog,
+					quotedSchema,
+					quotedTable
+				),
 				dialect
 		);
```

Comment:
I think instead of creating a new `QualifiedTableName` from the `TableInformation` may be enough to use the `Table#getQualifiedTableName() `.
```
final String tableName = jdbcEnvironment.getQualifiedObjectNameFormatter().format(
				getQualifiedTableName(),
				dialect
		);
```